Doc/library/sqlite3.rst

```diff
@@ -829,35 +829,55 @@ Exceptions
 
 .. exception:: Warning
 
-   A subclass of :exc:`Exception`.
+   Exception raised for important warnings like data truncations while
+   inserting, etc.  It is a subclass of :exc:`Exception`.
 
 .. exception:: Error
 
-   The base class of the other exceptions in this module.  It is a subclass
-   of :exc:`Exception`.
+   The base class of the other error exceptions in this module.  Use this to
+   catch all errors with one single :keyword:`except` statement.  :exc:`Error`
+   is a subclass of :exc:`Exception`.
+
+.. exception:: InterfaceError
+
+   Exception raised for errors that are related to the database interface
```

Comment:
Currently it is raised in two cases:

1. Some (not all) errors in binding parameters.
2. Attempt to fetch from cursor after commit/rollback.

I am not sure that these cases are related. And I am not sure that InterfaceError should be used in the first case (maybe only for unsupported types, but not for internal memory error and too long strings/blobs?).

Reply:
The way I read PEP 249, is that `SQLITE_MISUSE` should be converted to `InterfaceError`.

> 1. Some (not all) errors in binding parameters.

Unsupported type errors should be `ProgrammingError`s, IMO.

> 2. Attempt to fetch from cursor after commit/rollback.

This is an artefact from the past. IMO we should no longer raise an error; we should let SQLite deal with this, because in most cases it is perfectly able to do so. If it fails, SQLite will raise an appropriate error.